src/applications/Job.java

```diff
@@ -49,10 +49,38 @@ public void setArrivalTime(int arrivalTime) {
     public int getId() {
         return id;
     }
-
+    /**
+     * put the next job on the machine queue.
+     * @param machineShopSimulator
+     * @param p // the machine to get
+     */
 	void putJobOnMachineQueue(MachineShopSimulator machineShopSimulator, int p) {
         Machine machine = machineShopSimulator.getMachine(p);
         machine.getJobQ().put(this);
 	}
 
+	/**
+	 * move theJob to machine for its next task
+	 * @param machineShopSimulator TODO
+	 * @param simulationResults TODO
+     * @param timeNow
+     * @param eList
+     * @param largeTime
+	 * @return false if no next task, true if has next task.
+	 */
+	boolean moveToNextMachine(MachineShopSimulator machineShopSimulator, SimulationResults simulationResults, int timeNow, EventList eList, int largeTime) {
```

Comment:
I might suggest that this might be going toward a long param list smell

Reply:
Yeah that might be a good point. I think I'm going to get the timeNow and largeTime instead of passing them.

Reply:
Those changes should be up now.